src/dist/notifications.rs

```diff
@@ -174,11 +174,20 @@ impl<'a> Display for Notification<'a> {
                 "removing stray hash found at '{}' in order to continue",
                 path.display()
             ),
-            SkippingNightlyMissingComponent(components) => write!(
-                f,
-                "skipping nightly which is missing installed component '{}'",
-                components[0].short_name_in_manifest()
-            ),
+            SkippingNightlyMissingComponent(components) => {
+                let long_name = if components[0].target != TargetTriple::from_host() {
```

Comment:
Maybe add a comment here to explain?

Reply:
Thinking about it, this is slightly confusing if only because on multiarch systems, `TargetTriple::from_host()` is not necessarily the same as the triple of the toolchain.  We might need to augment this particular notification with the toolchain's host triple.

Reply:
So, you want to take the (user-defined) default toolchain triple as the base, instead of the current host? Or did I misunderstood you?

Reply:
The point at which you create this notification has access to a `Toolchain` instance which knows the host triple of the toolchain.  It'd be best to pass this into the notification so that it can format based on that, rather than on the host's triple.

Reply:
@kinnison Is f7e1ea4 what you meant?